website/project/views/contributor.py

```diff
@@ -127,8 +127,8 @@ def get_contributors_from_parent(auth, node, **kwargs):
         raise HTTPError(http.FORBIDDEN)
 
     contribs = [
-        profile_utils.add_contributor_json(contrib)
-        for contrib in parent.visible_contributors
+        profile_utils.add_contributor_json(contrib, node=node)
+        for contrib in parent.contributors if contrib not in node.visible_contributors
```

Comment:
Should this be `node.visible_contributors` or `node.contributors`?

Reply:
The idea of the ticket as I understand it was that you would import non-bibliographic (or "invisible") contributors as well as visible contributors, so importing all contributors not just visible ones was key.

Reply:
Yup, but this looks like, if a contributor is in the parent and in the node, _but_ it's in the node as a non-bibliographic contributor, it will try to add it again.

Reply:
Whereas you'd want to avoid re-adding whether they were bibliographic or no.

Reply:
(You should probably should add a test or two for these cases, I suspect)

Reply:
Right you are, I'll change that.